doc/specs/#120,121,119-Update-Uninstall-List.md

```diff
@@ -0,0 +1,324 @@
+
+#### Update --all Experience
+
+The Update --all will automatically being updating all available updates. The user will not get additional prompting. This will allow for easy scripting.
+
+#### Update app when no update is available
+
+If the user attempts to update an app that there is no known update for, the package manager will issue an error.  
+*There is no update available from the community repository. If you need to reinstall or repair the app, try `winget repair appname`.*
+
+#### Update No packages found
+
+Like with the install command, if winget cannot determine a specific package based on the query string, winget will error and ask for more specifics.  
+*No package found matching input criteria. Type winget list to see available applications.*
+
+#### Update -m, --manifest option
+
+When the manifest option is passed to the Update command, it works just like with installer. If the version is less than the currently installed application, it will launch the Update process.
+If the application is not already installed, this will generate an error.  
+*The application "appname" is not installed, therefore no updates are available. If you would like to install the app, use the install command.*
+
+#### Dependencies
+
+Handling dependencies will be a challenge once Windows Package Manager supports dependencies. Dependencies are not covered as part of this spec.
+
+> Open issue: If the application is not side by side, it will force an uninstall of the previous version to Update. We will need to assume the installer wil handle.
+
+### Uninstall
+
+The `uninstall` command is designed to uninstall an application. The uninstall command will be executed based on the string in the registry when it comes to ProductCode or Remove-AppPackage when it comes to MSIX and Store apps.
+
+#### Uninstall Usage
+
+`winget uninstall [[-q] <query>] [<options>]`
+
+![uninstall command](images\120-uninstall.png)
+
+| Argument | Description |
+|--------------|-------------|
+| `-q,--query` | The query used to search for an application. |
+| `-?, --help` | Gets additional help on this command. |
+
+#### Uninstall Options
+
+The following options are available.
+
+| Option | Description |
+|--------------|-------------|
+| `-m,--manifest` | The path to the manifest of the package. |
+| `--id` | Filter results by ID. |
+| `--name` | Filter results by name. |
+| `--moniker` | Filter results by application moniker. |
+| `--tag` | Filter results by tag
+| `--command` | Filter results by command
+| `-s,--source` | Find the application using the specified [source](source.md). |
+| `-e,--exact` | Find the application using exact match. |
+
+#### Uninstall Command
+
+Determining the uninstall string will be based on the type of installer. For EXEs and MSIs the command will be shellexec. If the app requires elevation, then the user will be prompted.
```

Comment:
I assume shellexec means using ShellExecute instead of CreateProcess? Is there a reason for that? ShellExecute requires a file path and arguments so the uninstall string would need to be parsed to split it, whereas CreateProcess can take the command line string directly which seems like a better fit.